core/error_test.go

```diff
@@ -10,12 +12,47 @@ func TestErrorStringFormat(t *testing.T) {
 	e := Errorf("%s", refStr)
 
 	fileName := "error_test.go"
-	lineNum := 10 // line number where error was formed
+	lineNum := 12 // line number where error was formed
+	funcName := "github.com/contiv/netplugin/core.TestErrorStringFormat"
 
-	expectedStr := fmt.Sprintf("%s [%s %d]", refStr, fileName, lineNum)
+	expectedStr := fmt.Sprintf("%s [%s %s %d]", refStr, funcName, fileName, lineNum)
 
 	if e.Error() != expectedStr {
 		t.Fatalf("error string mismatch. Expected: %q, got %q", expectedStr,
 			e.Error())
 	}
 }
+
+func getError(msg string) *Error {
+	return Errorf(msg)
+}
+
+func TestErrorStackTrace(t *testing.T) {
+	msg := "an error"
+	e := getError(msg)
+
+	if e.desc != msg {
+		t.Fatal("Description did not match provided")
+	}
+
+	fileName := "error_test.go"
+	lineNum := 27 // line number where error was formed
+	funcName := "github.com/contiv/netplugin/core.getError"
+
+	expectedStr := fmt.Sprintf("%s [%s %s %d]", msg, funcName, fileName, lineNum)
+
+	if e.Error() != expectedStr {
+		t.Fatalf("Error message yielded an incorrect result with CONTIV_TRACE unset: %s", e.Error())
+	}
+
+	os.Setenv("CONTIV_TRACE", "1")
```

Comment:
*minor comment*: may be we should reset this environment variable at the end of the test. Just so in future if we add more tests for this package it doesn't interfere with them.